circuit_knitting_toolbox/circuit_cutting/qpd/qpd.py

```diff
@@ -100,6 +106,57 @@ def generate_qpd_samples(
     return {k: (v, WeightType.SAMPLED) for k, v in random_samples.items()}
 
 
+def decompose_qpd_instructions(
+    circuit: QuantumCircuit,
+    instruction_ids: Sequence[Sequence[int]],
+    map_ids: Sequence[int] | None = None,
+) -> QuantumCircuit:
+    r"""
+    Replace all QPD instructions in the circuit with local Qiskit operations and measurements.
+
+    Args:
+        circuit: The circuit containing QPD instructions
+        instruction_ids: A 2D sequence, such that each inner sequence corresponds to indices of instructions comprising
+            one decomposition in the circuit. The elements within a common sequence belong to a common decomposition
+            and should be sampled together.
+        map_ids: Indices to a specific linear mapping to be applied to the decompositions
+            in the circuit. If no map IDs are provided, the circuit will be decomposed randomly
+            according to the decompositions' joint probability distribution.
+
+    Returns:
+        (:class:`~qiskit.QuantumCircuit`): Circuit which has had all its QPDGates decomposed into local operations.
+        The circuit will contain a new, final classical register to contain the QPD measurement
+        outcomes (accessible at ``retval.cregs[-1]``).
+
+    Raises:
```

Comment:
This raises an error if an instruction acting on a number of qubits not equal to 1 or 2. Should add to docstring